backend/src/main/java/ai/verta/modeldb/experiment/subtypes/CreateExperimentHandler.java

```diff
@@ -0,0 +1,291 @@
+  private final ArtifactHandler artifactHandler;
+  private final CodeVersionHandler codeVersionHandler;
+
+  public CreateExperimentHandler(
+      Executor executor,
+      FutureJdbi jdbi,
+      Config config,
+      UAC uac,
+      AttributeHandler attributeHandler,
+      TagsHandler tagsHandler,
+      ArtifactHandler artifactHandler) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.config = config;
+    this.uac = uac;
+
+    this.attributeHandler = attributeHandler;
+    this.tagsHandler = tagsHandler;
+    this.artifactHandler = artifactHandler;
+    this.codeVersionHandler = new CodeVersionHandler(executor, jdbi, "experiment");
+  }
+
+  public InternalFuture<Experiment> convertCreateRequest(
+      final CreateExperiment request, UserInfo userInfo) {
+    return InternalFuture.completedInternalFuture(getExperimentFromRequest(request, userInfo));
+  }
+
+  /**
+   * Method to convert createExperiment request to Experiment object. This method generates the
+   * experiment Id using UUID and puts it in Experiment object.
+   *
+   * @param request : CreateExperiment
+   * @return Experiment
+   */
+  private Experiment getExperimentFromRequest(CreateExperiment request, UserInfo userInfo) {
+
+    String errorMessage = null;
+    if (request.getProjectId().isEmpty()) {
+      errorMessage = "Project ID not found in CreateExperiment request";
+    } else if (request.getName().isEmpty()) {
+      request = request.toBuilder().setName(MetadataServiceImpl.createRandomName()).build();
+    }
+
+    if (errorMessage != null) {
+      throw new InvalidArgumentException(errorMessage);
+    }
+
+    /*
+     * Create Experiment entity from given CreateExperiment request. generate UUID and put as id in
+     * Experiment for uniqueness.
+     */
+    var experimentBuilder =
+        Experiment.newBuilder()
+            .setId(UUID.randomUUID().toString())
+            .setProjectId(request.getProjectId())
+            .setName(ModelDBUtils.checkEntityNameLength(request.getName()))
+            .setDescription(request.getDescription())
+            .addAllAttributes(
+                request.getAttributesList().stream()
+                    .sorted(Comparator.comparing(KeyValue::getKey))
+                    .collect(Collectors.toList()))
+            .addAllTags(TagsHandlerBase.checkEntityTagsLength(request.getTagsList()))
+            .addAllArtifacts(
+                request.getArtifactsList().stream()
+                    .sorted(Comparator.comparing(Artifact::getKey))
+                    .collect(Collectors.toList()))
+            .setVersionNumber(1L);
+
+    if (request.getDateCreated() != 0L) {
+      experimentBuilder
+          .setDateCreated(request.getDateCreated())
+          .setDateUpdated(request.getDateCreated());
+    } else {
+      experimentBuilder
+          .setDateCreated(Calendar.getInstance().getTimeInMillis())
+          .setDateUpdated(Calendar.getInstance().getTimeInMillis());
```

Comment:
Calendar isn't deprecated, but it is clearer to us `Instant`.  Also, 
```suggestion
          .setDateCreated(Instant.now().toEpochMilli())
          .setDateUpdated(Instant.now().toEpochMilli());
```

Also, do we want to allow the created time to be updated?

Reply:
No it is not deprecated. but we can take `Instant.now().toEpochMilli()` instead of Calendar

Reply:
Fixed